flujo/domain/dsl/pipeline_step_validations.py

```
@@ -269,18 +290,102 @@ def _validate_pipeline(
                     try:
                         child_paths = _validate_pipeline(
-                            branch, set(available_roots), set(produced_paths), None, None
+                            branch,
+                            set(branch_available_roots),
+                            set(branch_produced_paths),
+                            None,
+                            None,
                         )
-                        parallel_outputs.update(child_paths)
+                        branch_outputs_by_name[branch_key] = set(child_paths)
                     except Exception:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)